clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -475,6 +492,7 @@ protected PutObjectResponseType doPutOperation(final PutObjectType request) thro
       } catch (Exception ex) {
         LOG.debug("Failed to fire reporting event for OSG object creation", ex);
       }
+      OSGUtil.setCorsInfo(request, response, bucket.getBucketUuid());
```

Comment:
CORS response headers are in the bucket and object operations that are most likely to be used in CORS traffic. For completeness, I would add this to the rest of the operations. However, I'm hoping to find a better way of doing this than adding this bit of code to every operation method we have! @euca-nightfury has a possible solution, from the overall pull request comments:
"Instead of changing every outbound handler, how about creating a new handler for CORS processing and add it to the outbound stages"
To be investigated.